src/StepperISR_due.cpp

```diff
@@ -1,401 +1,691 @@
-        PWM_INTERFACE->PWM_CH_NUM[mapping->channel].PWM_CPRDUPD = e->ticks; 
-        return;
-      }
-      if (e->toggle_dir) {
-        e->toggle_dir = 0;
-        *_dirPinPort ^= _dirPinMask;
-        delay(50);
-      }
-    } else {
-      if (e->toggle_dir) {
-        e->toggle_dir = 0;
-        *_dirPinPort ^= _dirPinMask;
-        delayMicroseconds(30);
-      } else {
-        switch (_queue_num) {
-        case 0:
-          InterruptMap[_queue_num].ISR = Queue0ISR;
-          attachPWMPeripheral(mapping->port, mapping->pin, mapping->channelMask,
-                              Queue0ISR);
-          break;
-        case 1:
-          InterruptMap[_queue_num].ISR = Queue1ISR;
-          attachPWMPeripheral(mapping->port, mapping->pin, mapping->channelMask,
-                              Queue1ISR);
-          break;
-        case 2:
-          InterruptMap[_queue_num].ISR = Queue1ISR;
-          attachPWMPeripheral(mapping->port, mapping->pin, mapping->channelMask,
-                              Queue2ISR);
-          break;
-        case 3:
-          InterruptMap[_queue_num].ISR = Queue3ISR;
-          attachPWMPeripheral(mapping->port, mapping->pin, mapping->channelMask,
-                              Queue3ISR);
-          break;
-        case 4:
-          InterruptMap[_queue_num].ISR = Queue4ISR;
-          attachPWMPeripheral(mapping->port, mapping->pin, mapping->channelMask,
-                              Queue4ISR);
-          break;
-        case 5:
-          InterruptMap[_queue_num].ISR = Queue5ISR;
-          attachPWMPeripheral(mapping->port, mapping->pin, mapping->channelMask,
-                              Queue5ISR);
-          break;
-        }
-        _hasISRactive = true;
-        _connected = true;
-      }
-    }
-  }
-}
-
-int8_t StepperQueue::startPreparedQueue() {
-  if (next_write_idx == read_idx) {
-    return AQE_ERROR_EMPTY_QUEUE_TO_START;
-  }
-  if (PWM_INTERFACE->PWM_SR & (1 << mapping->channel)) {
-    noInterrupts();
-    struct queue_entry* e = &entry[read_idx & QUEUE_LEN_MASK];
-    interrupts();
-    //That should be it...We should start the next command on the next cycle
-    PWM_INTERFACE->PWM_CH_NUM[mapping->channel].PWM_CPRDUPD = e->ticks; 
-    _hasISRactive = true;
-    return AQE_OK;
-  }
-  return AQE_OK;
-}
-
-void StepperQueue::forceStop() {
-  read_idx = next_write_idx;
-  PWMC_DisableChannel(PWM_INTERFACE, mapping->channel);
-}
-
-bool StepperQueue::isValidStepPin(uint8_t step_pin) {
-  if (pinToChannel(step_pin) < 0x08) {
-    if (!channelsUsed[pinToChannel(step_pin)]) {
-      channelsUsed[pinToChannel(step_pin)] = true;
-      numChannels++;
-      return true;
-    }
-  }
-  return false;
-}
-
-bool StepperQueue::isRunning() {
-  if (_hasISRactive) {
-    return true;
-  }
-}
-
-int8_t StepperQueue::queueNumForStepPin(uint8_t step_pin) { return -1; }
-#endif
\ No newline at end of file
+#include "StepperISR.h"
+#ifndef ARDUINO_ARCH_SAM
+#define ARDUINO_ARCH_SAM
+#endif
```

Comment:
This definition of ARDUINO_ARCH_SAM causes for the other architectures errors for double defined functions. Any reason to keep this ?

Reply:
Ah, now that was an old line I completely forgot about.  One quirck of Visual Studio is that it doesn't pick up compiler definitions, so without the #define, all the code I was working on was greyed out, no autocomplete, no intellisense, etc.  So yes, I should not be there.  I had thought I'd remember it, but the time away from this caused me to easily forget it.